packages/amplify_core/lib/src/types/api/rest/rest_exception.dart

```diff
@@ -19,16 +19,8 @@ import 'package:amplify_core/amplify_core.dart';
 /// An HTTP error encountered during a REST API call, i.e. for calls returning
 /// non-2xx status codes.
 /// {@endtemplate}
-class RestException extends ApiException {
-  /// The HTTP response from the server.
-  final RestResponse response;
-
+@Deprecated('No longer thrown for non-200 responses. Will soon be removed')
```

Comment:
If a customer was using this to do something on non-200 responses they will have to make code changes if they want their app to function the same way, correct?

I'm on the fence about this, but I think we should consider removing this rather than marking it as deprecated. Marking it as deprecated seems misleading to me. I have always thought of the deprecation annotation as an indication that something *will* break. Not that it has already been broken.

I'm on the fence because 1) the message could be helpful, and 2) (I think) some customers will be using `RestException` for other reasons than catching non-200 responses.

If we keep the deprecation annotation, I think the message needs to make it clear that the app may not function as intended until the code is removed. Something like, "BREAKING CHANGE: RestException is no longer thrown for non-200 responses. See the breaking changes section of the latest release notes for migration instructions."

Here is what the Dart docs say about the [Deprecated annotation](https://api.flutter.dev/flutter/dart-core/Deprecated-class.html) FWIW

> The intent of the @Deprecated annotation is to inform authors who are currently using the feature, that they will soon need to stop using that feature in their code, even if the feature is currently still working correctly.
> 
> Deprecation is an early warning that the deprecated feature is scheduled to be removed at a later time, a time possibly specified in [message](https://api.flutter.dev/flutter/dart-core/Deprecated/message.html). A deprecated feature should no longer be used, code using it will break at some point in the future. If existing code is using the feature, that code should be rewritten to no longer use the deprecated feature.

Reply:
Agree we are not using it exactly as intended according to that blurb. I took your suggestion of changing the deprecation message to be a little more grabby given that we are kinda hacking the deprecation message to communicate why it's not there. I think it would be tough to use it more traditionally here because the old response type is a field on the exception, which is what started this thought process. I think we could consider a deprecation on main/stable as this moves toward GA.